client/src/main/java/io/pravega/client/stream/impl/ReaderGroupImpl.java

```diff
@@ -200,8 +210,144 @@ private Checkpoint completeCheckpoint(String checkpointName) {
 
     @Override
     public void resetReaderGroup(ReaderGroupConfig config) {
+        while (true) {
+            val state = getState();
+            switch (state.getConfigState()) {
+                case INITIALIZING:
+                    doInit(state);
+                    break;
+                case REINITIALIZING:
+                    doReinit(state);
+                    break;
+                case READY:
+                    long newGen = state.getGeneration() + 1;
+                    // If true then at this point the update is completed and it is in the desired state (REINITIALIZING)
+                    // on which doReinit is called to continue the reset with this newState.
+                    if (doStateTransition(state, new ReaderGroupStateResetStart(config, newGen))) {
+                        val newState = getState();
+                        doReinit(newState);
+                        return;
+                    }
+                    break;
+                case DELETING:
+                    doDelete(state);
+                    throw new ReinitializationRequiredException("This ReaderGroup does not exist.");
+            }
+        }
+    }
+
+    public void createState(ReaderGroupConfig config) {
```

Comment:
Why are these two public methods `createState`  and `deleteState` not part of the implemented interface?

Reply:
The methods that we have as not part of the implemented interface were because we did not want the user to be able to see them. But we still needed access to the `ReaderGroupImpl` to perform these operations. So we decided we would have these as only a part of `ReaderGroupImpl` and not `ReaderGroup`. So to access them we cast a `ReaderGroup` object to `ReaderGroupImpl`.